endpoint.go

```diff
@@ -549,3 +554,48 @@ func (ep *endpoint) DataScope() datastore.DataScope {
 func (ep *endpoint) isLocalScoped() bool {
 	return ep.DataScope() == datastore.LocalScope
 }
+
+func (ep *endpoint) assignAddress() (err error) {
+	var ipa ipamapi.Allocator
+	n := ep.getNetwork()
+	if n.Type() == "host" || n.Type() == "null" || n.Type() == "bridge" {
+		return
+	}
+	ipa, err = n.ctrlr.getIPAllocator(n.ipamType)
+	if err != nil {
+		return
+	}
+	n.Lock()
```

Comment:
You cannot hold the lock while invoking `ipa.RequestAddress`. Remember `RequestAddress` can be a remote call